ls1intum/Artemis
# Extract Spring Beans for Programming Exercise Connectors

### Checklist

### Description 
These are the production code changes necessary for #1168 .
Please refer to the above mentioned PR for the documentation .

These changes are necessary, because we never really had a full integration test in mind when implementing the connector services. This resulted in a "non-Spring-friendly" implementation, meaning we currently instantiate a lot of objects directly in methods and don't mock them, or just don't offer the possibility to mock outgoing requests. More detailed information can be found in #1168 

### How To Test
1. Create a new programming exercise
2. Edit the exercise
3. Participate in the exercise

Please make sure that everything works as before!
